src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -543,7 +550,17 @@ private async Task CheckJobCompletionStatus()
                 // all queries marked as complete, reindex job is done, check success or failure
                 if (_reindexJobRecord.QueryList.Keys.All(q => q.Status == OperationStatus.Completed))
                 {
-                    await UpdateParametersAndCompleteJob();
+                    // Perform a final check to make sure there are no resources left to reindex
+                    int totalCount = await CalculateTotalCount();
+                    if (totalCount != 0)
+                    {
+                        string message = $"{totalCount} resources failed to be reindexed.";
+                        await HandleRetryableError(message, OperationOutcomeConstants.IssueType.Incomplete);
```

Comment:
Is the intent here to retry this total count again? Or is this error not something we expect to be able to recover from on a retry?

Reply:
As discussed, it is difficult to know if we could recover from this on a retry, since this failure will ultimately never be reached!

@Ivanidzo4ka made a good point below that automatically retrying a process that could take many days might not be the best user experience. It would instead be favourable to give control to the user to decide if/when to retry the reindexing operation.

I removed the retry logic in f1bce27 and updated [AB#80638](https://microsofthealth.visualstudio.com/Health/_workitems/edit/80638).